rust/worker/src/execution/operators/apply_log_to_segment_writer.rs

```diff
@@ -171,56 +166,25 @@ impl Operator<WriteSegmentsInput, WriteSegmentsOutput> for WriteSegmentsOperator
             Ok(records) => records,
             Err(e) => {
                 tracing::error!("Error materializing records {}", e);
-                return Err(WriteSegmentsOperatorError::LogMaterializationError(e));
+                return Err(ApplyLogToSegmentWriterOperatorError::LogMaterializationError(e));
             }
         };
         // Apply materialized records.
         match input
-            .record_segment_writer
-            .apply_materialized_log_chunk(res.clone())
-            .instrument(tracing::trace_span!(
-                "Apply materialized logs to record segment"
-            ))
-            .await
-        {
-            Ok(()) => (),
-            Err(e) => {
-                return Err(WriteSegmentsOperatorError::ApplyMaterializatedLogsError(e));
-            }
-        }
-        tracing::debug!("Applied materialized records to record segment");
-        match input
-            .metadata_segment_writer
+            .segment_writer
             .apply_materialized_log_chunk(res.clone())
             .instrument(tracing::trace_span!(
-                "Apply materialized logs to metadata segment"
+                "Apply materialized logs",
+                segment = input.segment_writer.get_name()
```

Comment:
unfortunately the span name cannot be dynamically created (could use a macro & const but didn't seem worth it)